bin/brew

```diff
@@ -45,5 +85,77 @@ fi
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_RUBY_PATH
+export HOMEBREW_PREFIX
+export HOMEBREW_REPOSITORY
+export HOMEBREW_LIBRARY
+export HOMEBREW_CELLAR
+
+if [ -n "$HOMEBREW_OSX" ]
+then
+  if [ "$(xcode-\select --print-path)" = "/" ]
+  then
+    cat >&2 <<EOS
```

Comment:
It's slightly more complex in a single place, but avoids repetitive and error-prone code in many different places, so that's my main motivation. (Because of that, I don't think it's merely stylistic. The complex part could be explained in a comment: “read `stdin` into a variable”.

I'd be interested to know if you disagree with that in general or if you basically agree, but don't want to change it for the sake of getting something merged.